src/editorutil.ts

```diff
@@ -4,11 +4,30 @@
  *-----------------------------------------------------------------------------------------------*/
 
 'use strict';
+import * as fs from 'fs';
+import { Protocol } from 'rsp-client';
+import { ServerExplorer } from './serverExplorer';
+import { Utils } from './utils';
 import * as vscode from 'vscode';
+const tmp = require('tmp');
 
 export class EditorUtil {
-    constructor() {
+
+    private static instance: EditorUtil;
+    private explorer: ServerExplorer;
+    private serverTmpFiles: Map<string, string> = new Map<string, string>();
+
+    constructor(explorer: ServerExplorer) {
```

Comment:
shouldn't we use a private constructor given that we're using it as singleton via ```getInstance```?